### PR TITLE
Remove laxtuple

### PR DESCRIPTION
**Pull request by fehiepsi**
The new version of JAX supports namedtuple out of box, so we can drop laxtuple for now. :)

The slowness of `np.stack/np.concatenate` is also solved. So we can use it in `fori_collect`.

### CODE REVIEW

File: examples/vae.py [hunk 1]
```diff
@@ -83,7 +83,6 @@ def main(args):
     opt_init, opt_update, get_params = optimizers.adam(args.learning_rate)
     svi_init, svi_update, svi_eval = svi(model, guide, elbo, opt_init, opt_update, get_params,
                                          encode=encode, decode=decode, z_dim=args.z_dim)
-    svi_update = jit(svi_update)
```
Comment (fehiepsi) [inline examples/vae.py]: this is not necessary because we use `svi_update` inside a jitted function.

File: numpyro/mcmc.py [hunk 2]
```diff
@@ -270,7 +260,8 @@ def _nuts_next(step_size, inverse_mass_matrix, vv_state, rng):
                                  max_tree_depth=max_treedepth)
         accept_prob = binary_tree.sum_accept_probs / binary_tree.num_proposals
         num_steps = binary_tree.num_proposals
-        vv_state = vv_state.update(z=binary_tree.z_proposal,
+        vv_state = IntegratorState(z=binary_tree.z_proposal,
+                                   r=vv_state.r,
```
Comment (neerajprad) [inline numpyro/mcmc.py]: Is this added for logging purposes?

**Comment by neerajprad** [comment]
Feel free to merge this at your convenience!

File: numpyro/mcmc.py (hunk 2 above)
Comment (fehiepsi) [reply numpyro/mcmc.py]: This is added because namedtuple does not have `update` method. Here, `r` field keeps its old value.

Comment (neerajprad) [reply numpyro/mcmc.py]: Ahh..sorry, I missed that. Makes sense.

**Pull request merged**